### PR TITLE
bradl3yC 2636 refactor direct deposit errors

### PR DESCRIPTION
**Pull request by bradl3yC**
## Description
Refactor code that parses Direct Deposit save errors
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/2636

## Current Behavior
Errors are evaluated individually for the key and then the text

## Desired Behavior
Evaluate both the key and the text to remove or lessen the possibility of false positives generated by looking at the key or text alone

## Testing done
Added unit tests to validate that both the key and text are both being evaluated for each error as opposed to independently (creates potential false positives)

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile360/tests/util/index.unit.spec.js [hunk 1]
```diff
@@ -103,4 +107,51 @@ describe('profile utils', () => {
...
+                text: 'Flashes on record',
+              },
+            ],
+          },
+        },
+      ];
+      expect(hasFlaggedForFraudError(errors)).to.equal(true);
+    });
+
+    it('hasInvalidHomePhoneNumberError returns false if text does not contain night phone', () => {
```
Comment (bradl3yC) [inline src/applications/personalization/profile360/tests/util/index.unit.spec.js]: This tests the possible false positive condition Erik described in the ticket.  We want to test the key and the text in the same pass instead of individually.

Comment (erikphansen) [reply src/applications/personalization/profile360/tests/util/index.unit.spec.js]: So the case I'm worried about, which admittedly not likely to happen because I don't know if we will ever get back multiple entries in that `meta.messages` array, is if the `meta.messages` array looked like this:
```
[
  {
    key: 'cnp.payment.generic.error.message',
    severity: 'ERROR',
    text: "some error we don't know about",
  },
  {
    key: 'unknown.key',
    severity: 'ERROR',
    text:
      'Generic CnP payment update error. Update response: Update Failed: Night area number is invalid, must be 3 digits',
  },
]
```
In this case the `hasInvalidHomePhoneNumberError()` should return `false` because although the `cnp.payment.generic.error.message` code appears on a message object and the `night area number` appears in a message object's `text`, they appear on _different_ message objects.

This test you wrote is totally valid but it doesn't test for this hypothetical false-positive I noticed we could get. Again, I'm wanting this to be coded overly defensively, but might as well do it right, right? :)

Basically the parsing logic needs to be smarter and check that both the `key` and `text` of _the same message object_ match what we are looking when checking which error message we should show.

**Comment by rianfowler** [comment]
Please tag the @department-of-veterans-affairs/frontend-review-group in a new comment and remove the "needs peer approval" label when this is has a peer review approval. 

We recommend using [draft PRs](https://github.blog/2019-02-14-introducing-draft-pull-requests/) for peer reviews and only promoting these into full PRs when they've received a peer approval: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/code_review_guidelines.md

File: src/applications/personalization/profile360/util/index.js [hunk 2]
```diff
@@ -24,82 +24,46 @@ export async function getData(apiRoute, options) {
...
+export const hasInvalidWorkPhoneNumberError = errors =>
+  hasErrorMessage(errors, GENERIC_ERROR_KEY, 'day phone number') ||
+  hasErrorMessage(errors, GENERIC_ERROR_KEY, 'day area number');
 
 // Helper that creates and returns an object to pass to the recordEvent()
 // function when an errors occurs while trying to save/update a user's direct
 // deposit payment information. The value of the `error-key` prop will change
 // depending on the content of the `errors` array.
-export function createDirectDepositAnalyticsDataObject(errors) {
+export const createDirectDepositAnalyticsDataObject = errors => {
```
[the remark this answers is not in the document]
Comment (bradl3yC) [reply src/applications/personalization/profile360/util/index.js]: @erikphansen when using them as an expression they're subject to hoisting.  This is an added layer of protection against undefines or lint errors and there is no cost difference when going this route.

File: src/applications/personalization/profile360/tests/util/index.unit.spec.js (hunk 1 above)
Comment (bradl3yC) [reply src/applications/personalization/profile360/tests/util/index.unit.spec.js]: @erikphansen actually, it does!  https://github.com/department-of-veterans-affairs/vets-website/pull/10951/files#diff-252aebd3806ab343cc3a5fc208865419R27 . I changed the method thats evaluating the errors and if the message text key/value is present, it evaluates it at the same time as the error key.

Comment (bradl3yC) [reply src/applications/personalization/profile360/tests/util/index.unit.spec.js]: @erikphansen I added a test case to cover this ^^

**Review by kelsonic — APPROVED** [review]
LGTM! Nice work! 🎉

**Review by erikphansen — APPROVED** [review]
D'oh! Sorry I misread your code. Thanks for adding that additional test!

**Comment by bradl3yC** [comment]
@department-of-veterans-affairs/frontend-review-group ready for review

**Review by rianfowler — APPROVED** [review]
